src/Functions/FunctionsConversion.h

```diff
@@ -1256,7 +1252,7 @@ struct ConvertThroughParsing
 
             if constexpr (exception_mode == ConvertFromStringExceptionMode::Throw)
             {
-                if constexpr (parsing_mode == ConvertFromStringParsingMode::BestEffort)
+                if constexpr (parsing_mode == ConvertFromStringParsingMode::BestEffort && (to_datetime || to_datetime64))
```

Comment:
why do we want to exclude `to_time` and `to_time64` here and further?

Reply:
Ah, types Time and Time64 were added recently and my branch didn't have them. They have to be included yes.